book/java-rebuilds-service.md

```diff
@@ -0,0 +1,154 @@
+# JVM Build Service
+
+## Overview
+
+The JVM Build Service (JBS) is a controller that will rebuild Java and other JVM Language based dependencies from source.
+
+Java uses a binary distribution model, so the only way to ensure an application is completly built from source is to rebuild all its component libraries from source themselves. Unfortunatly due to the nature of the Java ecosystem this is not an easy process and would normally require a large amount of manual toil.
+
+JBS will attempt to automate this proccess as much as possible, making it much less time consuming to create fully supply chain controlled Java builds.
+
+### Dependencies
+
+JBS depends on Tekton, but is otherwise largely independent of the rest of RHTAP. Given the right configuration it can be deployed outside of RHTAP, which is mainly used for testing and development.
+
+## Architecture
+
+### Flow Overview
+
+The JVM Build service can be enabled on a specific namespace by creating a `JBSConfig` object, with `enableRebuilds: true` specified. This will trigger the controller to create a deployment of the local Cache in the namespace. This cache has a number of purposes, but the primary one is to cache artifacts from Maven central to speed up Java builds.
```

Comment:
Can you provide a sample object?

Reply:
I have added it towards the bottom of the document.